### DISCUSSION TITLE
TinyMediaManager & it's 3rd Party Scraper Addon

### DISCUSSION THREAD

**Discussion by ms52538** [CodyBerenson/PGMA-Modernized#174] [Ideas]
Has there been any thought given to code development to enable Modernize' plugins to work with other tools, such as TinyMediaManager? I use TinyMM for my non-adult content; building related content for each movie and then utilizing PLEX's local meta content agents to scrape and populate libraries for cinema and tv. I was wondering if something similar could be facilitated for the modernized scraper plugins. I'm uncertain if TinyMM's third party addon would be able to facilitate handshaking with Modernize's scrapers, but the app does seem to have a 3rd party development interface for such things: https://www.tinymediamanager.org/blog/third-party-scraper-addons/. I could foresee an immediate benefit of such a partnership would be placing localized meta content into our collections for PLEX localized meta matching AND the ability to directly scrape if needed. Basically two options for meta collection and management (vs one).  Perhaps this has already been discussed before but I could not find anything referencing TinyMM in previous conversations. Thoughts?

**Comment by CodyBerenson** [comment]
For a couple years now, Jason and I have been discussing moving to a NFO based approach, where our scrapers are consumed by a tool such as TinyMM to 1) scrape index(es) to create/populate .NFO and 2) move the media file, the newly created NFO, and the acquired poster to a new folder.  There are already straight adult versions (e.g., JAVMovieScraper) that allow you to choose from a number of scraping sites.  The only real holdups have been 1) bandwidth and 2) fluency in JAVA.  

IMO, whether it is TinyMM or something homegrown in python, going to a NFO based approach is absolutely the way to go...especially if/when a library needs to be recreated (so it won't take days and days and result in Denial of Service responses from the Indexing sites).   TinyMM along with our agents retrofitted as third party scraper addons is exactly the use case I'd envisioned.....it will just take some JAVA skills.  

Thanks for starting this thread.  JAVA anyone?
